Algorithms/Search/KmpSearcher.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections.Generic;
+using Utilities.Exceptions;
+
+namespace Algorithms.Search
+{
+    /// <summary>
+    /// An implementation of Knuth–Morris–Pratt Algorithm.
+    /// Worst case time complexity: O(n).
+    /// </summary>
+    public class KmpSearcher
```

Comment:
Please extend this name to include full surnames

Reply:
Resolved.
661bd9a